Silicon/CommonSocPkg/Library/GpioV2Lib/GpioV2Init.c

```diff
@@ -0,0 +1,1635 @@
+      break;
+    case GpioV2GpiGpeEnReg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.GpiGpeEn;
+      break;
+    case GpioV2SmiStsReg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.SmiSts;
+      break;
+    case GpioV2SmiEnReg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.SmiEn;
+      break;
+    case GpioV2NmiStsReg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.NmiSts;
+      break;
+    case GpioV2NmiEnReg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.NmiEn;
+      break;
+    case GpioV2Dw0Reg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.Dw0 + PadIndex * 0x10;
+      break;
+    case GpioV2Dw1Reg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.Dw0 + 0x04 + PadIndex * 0x10;
+      break;
+    case GpioV2Dw2Reg:
+      *RegisterOffset = MtlSocGpioGetGroups(CommunityIndex,GroupIndex).RegisterOffsets.Dw0 + 0x08 + PadIndex * 0x10;
+      break;
+    case GpioV2MiscCfg:
+      *RegisterOffset = MtlSocGpioGetCommunities(CommunityIndex)->RegisterOffsets.MiscCfg;
+      break;
+    case GpioV2AcReg:
+      *RegisterOffset = MtlSocGpioGetCommunities(CommunityIndex)->RegisterOffsets.AcSaiGrup0RcpDw0;
+      break;
+    default:
+      ASSERT(FALSE);
+      return EFI_INVALID_PARAMETER;
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+  This procedure will set GPIO mode
+
+  @param[in] GPIOV2_SERVICES      Gpio Services
+  @param[in] GpioPad              Gpio Pad. Please refer to GpioPinsYYY.h - where YYY name of the platform (eg. MTL, EBG, ...)
+  @param[in] PadModeValue         GPIO pad mode value
+
+  @retval EFI_SUCCESS             The function completed successfully
+  @retval EFI_INVALID_PARAMETER   Invalid group or pad number
+**/
+EFI_STATUS
+EFIAPI
+SetPadMode (
+  IN GPIOV2_PAD              GpioPad,
+  IN GPIOV2_PAD_MODE         PadMode
+  )
+{
+  UINT32  AndValue;
+  UINT32  OrValue;
+  UINT32  CommunityIndex;
+  UINT32  RegisterOffset;
+  BOOLEAN PadValid;
+  P2SB_SIDEBAND_REGISTER_ACCESS   P2sbPtr;
+
+  IsPadValid (GpioPad, &PadValid);
+  if (PadValid != TRUE) {
+    ASSERT (FALSE);
+    return EFI_INVALID_PARAMETER;
+  }
+
+  CommunityIndex = GPIOV2_PAD_GET_COMMUNITY_INDEX (GpioPad);
+
+
+  if (CommunityIndex == 0) {
+    P2sbPtr = *((P2SB_SIDEBAND_REGISTER_ACCESS*)PatchPcdGet32(PcdP2sb0Ptr));
+  } else if (CommunityIndex == 1) {
+    P2sbPtr = *((P2SB_SIDEBAND_REGISTER_ACCESS*)PatchPcdGet32(PcdP2sb1Ptr));
+  } else if (CommunityIndex == 2) {
+    P2sbPtr = *((P2SB_SIDEBAND_REGISTER_ACCESS*)PatchPcdGet32(PcdP2sb2Ptr));
+  } else if (CommunityIndex == 3) {
+    P2sbPtr = *((P2SB_SIDEBAND_REGISTER_ACCESS*)PatchPcdGet32(PcdP2sb3Ptr));
+  } else if (CommunityIndex == 4) {
+    P2sbPtr = *((P2SB_SIDEBAND_REGISTER_ACCESS*)PatchPcdGet32(PcdP2sb4Ptr));
+  }
```

Comment:
This logic is called almost in every function. better to create a separate function to do it.

Reply:
I've created a separate function for this, thanks.